### PR TITLE
[MIG][10.0]product_supplier_code_purchase

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
supersedes #463

### CODE REVIEW

File: product_supplier_code_purchase/models/purchase_order.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    def _compute_product_supplier_code(self):
+        for line in self:
+            supplier_info = line.product_id.seller_ids.filtered(
+                lambda s: (s.product_id == line.product_id and
+                           s.name == line.partner_id))
+            if supplier_info:
+                code = supplier_info.product_code or ''
+                line.product_supplier_code = code
+
+    product_supplier_code = fields.Char(
```
Comment (rousseldenis) [inline product_supplier_code_purchase/models/purchase_order.py]: @aheficent Put fields definition on top. So, you can make
```python
compute='_compute_product_supplier_code'
```

File: product_supplier_code_purchase/models/purchase_order.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    @api.multi
+    @api.depends('product_id.seller_ids.name',
```
Comment (rousseldenis) [inline product_supplier_code_purchase/models/purchase_order.py]: @aheficent Make it depends on partner_id too.

File: product_supplier_code_purchase/models/purchase_order.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                 'product_id.seller_ids.product_code')
+    def _compute_product_supplier_code(self):
+        for line in self:
+            supplier_info = line.product_id.seller_ids.filtered(
```
Comment (rousseldenis) [inline product_supplier_code_purchase/models/purchase_order.py]: @aheficent Careful with seller_ids as you can have several lines for a product with the same partner (supplierinfo is also supplier pricelist).
This code will crash if several.

You should maybe take supplierinfo[0].

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Minor changes

**Comment by AaronHForgeFlow** [comment]
Pending tests still

**Review by rousseldenis — APPROVED** [review]

**Review by MiquelRForgeFlow — APPROVED** [review]

**Review by LoisRForgeFlow — APPROVED** [review]
:+1:

File: product_supplier_code_purchase/models/purchase_order.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    product_supplier_code = fields.Char(
+        string='Product Supplier Code',
+        compute='_compute_product_supplier_code'
```
Comment (legalsylvain) [inline product_supplier_code_purchase/models/purchase_order.py:14]: Should be great to have a function inv defined, to have the possibility, from the purchase order, to change the supplier code defined in the supplierinfo.

don't you think ? 

regards.

**Comment by AaronHForgeFlow** [comment]
@legalsylvain that may be useful in some cases. I'm not used to inverse methods. When the user changes the supplier code in the PO line would that always change the code in the supplier.info?

**Comment by legalsylvain** [comment]
It's just an idea !

But well : 

-> I think it's very usefull for the users, when the code is not defined, to set directly it in the current purchase order, instead of opening the product, and then the supplierinfo. (usability improved, reducing click quantity)

-> If the code is not set, (or wrong), changing it in the purchase order should update it in the product supplierinfo. I don't see any use case when the user doesn't want to update the code defined in the product.

**Comment by AaronHForgeFlow** [comment]
Yes I think that would be a good improvement, avoiding many mouse clicks.

**Pull request merged**